### PR TITLE
LIIKUNTA-252 | Show opening hours from tprek connections

### PR DESCRIPTION
**Pull request by Jokinen**
## Description

Shows opening hours for venue from tprek's connections.

Replaces opening hours found from hauki in the sidebar with ones found from tprek.

First 10 lines are shown by default. User can select expand and show all lines.

Adds is open now/Opens later info text into header that's based off of Hauki data.

### Closes

**[LIIKUNTA-252](https://helsinkisolutionoffice.atlassian.net/browse/LIIKUNTA-252)**

### Automated tests

Adjusted tests to cover new behaviour.

### Manual testing

- For testing hauki times: /paikat/tprek:358
- For testing tprek opening hours: /paikat/tprek:41102

## Screenshots

Screenshot 2022-01-19 at 6 36 26
Screenshot 2022-01-19 at 6 36 36
Screenshot 2022-01-19 at 6 45 02

### CODE REVIEW

**Review by Sartonon — COMMENTED** [review]
Looks good overall 👍  Added couple noted

File: src/common/utils/time/humanizeOpeningHoursForWeek.ts [hunk 1]
```diff
@@ -129,6 +129,13 @@ function humanizeTimeResourceState(
   }
 }
 
+export function humanizeTimeResourceState(
+  resourceState: TimeResourceState,
+  locale: Locale
+): string {
+  return _humanizeTimeResourceState(resourceState, microCopy[locale]);
+}
```
[the remark this answers is not in the document]
Comment (Jokinen) [reply src/common/utils/time/humanizeOpeningHoursForWeek.ts:137]: I've marked the other as a private implementation with an underscore. It allowed me to provide an API outwards that doesn't expose `microCopy` while also leaving the internal APIs as they are. In time the naming may diverge. I spent some time on this, but as it is such a minor detail, I moved onwards after a while. If the underscore pattern isn't up-to-date anymore, I could add a comment which further explains that this functions exists to differentiate the public and private API.

File: src/domain/venues/utils/getVenueOpeningTimeDescriptor.ts [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+}
+
+function getNextOpeningHoursFromNow(openingHours: OpeningHour[]): Time | null {
+  const now = new Date();
+  const todayDate = format(new Date(), "yyyy-MM-dd");
+
+  const openingHoursToday = openingHours.find(
+    (openingHour) => openingHour.date === todayDate
+  );
+  const nextTimeToday = openingHoursToday?.times?.reduce((incumbent, time) => {
```
[the remark this answers is not in the document]
Comment (Jokinen) [reply src/domain/venues/utils/getVenueOpeningTimeDescriptor.ts:53]: I looked at the function, but couldn't confidently determine what lines could be extracted. Do you have any suggestions?

My concern is that extracting some lines could actually make the function even harder to understand as you would need to jump around in the file. For instance you could extract the lines that find `nextTimeToday` and `nextTimeOtherDays` and you'd make the function easier to reason about in one way, but harder in another which could result in a net gain of very little.

[the remark this answers is not in the document]
Comment (Sartonon) [reply src/domain/venues/utils/getVenueOpeningTimeDescriptor.ts:53]: I agree that extracting new functions doesn't necessarily improve readability. If you feel like there is no easy way to make this more understandable, I'm ok with the current solution

**Review by Sartonon — APPROVED** [review]
Great 👍

**Pull request merged**